ui/src/components/ArbDataTable/index.js

```diff
@@ -0,0 +1,77 @@
+import React, { useState, useRef } from 'react';
+import Table from 'react-bootstrap/Table';
+
+const columns = [
+	'market',
+	'coin',
+	'from',
+	'to',
+	'buy',
+	'sell',
+	'profit'
+];
+
+function ArbDataTable({
+	from,
+	to,
+	amount,
+	market
+}) {
+	const [data, setData] = useState([]);
+    const [error, setError] = useState(null);
+    const intervalRef = useRef(null);
+
+    React.useEffect(() => {
+        if(intervalRef.current) {
+            clearInterval(intervalRef.current);
+        }
+        fetchData(from, to, amount, market);
+        intervalRef.current = setInterval(() => { 
+            fetchData(from, to, amount, market);
+        }, 5000);
+    }, [from, to, market, amount])
+
+	const fetchData = async (from, to, amount, market) => {
+        try {
+            const response = await fetch(`http://localhost:9999/arbitrage?from=${from}&to=${to}&amount=${amount}&market=${market}`);
+            const body = await response.json();
+            if (response.status === 200) {
+                setData(body);
```

Comment:
What if error was null prior to this, will lines 39 and 40 call render twice? or does react have some mechanism to batch updates? 

This is more of a question than a comment/review since I want to know.

Reply:
If we set the same value to a state, react doesn't re-render it. Also we are not using the `error` state variable anywhere to conditionally render. It is only defined. So `setError` will never cause a re-render

Reply:
I know that same value will not cause a re-render. What is the error was NOT null and set error was called right after (or before) set data? This is not specific to what you have done, it is a question directed at how react does things. 

For instance, we have a UI that shows a table or a error banner on top of the table. Let's assume our initial state is showing the error banner, now if we call setData{which renders the table} and then setError(null) {this hides the table}. Can we have a brief moment where the error banner and the table are shown together? or does react batch state updates?

Reply:
Good question. I don't know if it does. Found this on internet
```
React batches state updates that occur in event handlers and lifecycle methods. Thus, if you update state multiple times in a <div onClick /> handler, React will wait for event handling to finish before re-rendering.

To be clear, this only works in React-controlled synthetic event handlers and lifecycle methods. State updates are not batched in AJAX and setTimeout event handlers, for example.
```

In this case it might not batch right according to the above? Also thinking about this particular case, Body and error will never exist together. How about we have one state variable which is an object like `{  body: data, error: null }` and we only set this state ?